### PR TITLE
[TS migration] Migrate 'MentionSuggestions.js' component to TypeScript

### PR DESCRIPTION
**Pull request by pasyukevich**
### Fixed Issues

$ https://github.com/Expensify/App/issues/25062
PROPOSAL: 

### Tests

1. Open any report
2. Start writing mention, check that block with suggestion works as expected 

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open any report
2. Start writing mention, check that block with suggestion works as expected 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/23215729/62928e82-d344-4274-ac0e-14803c10a60c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/23215729/c39ec0a2-155d-4d39-bbde-e323f4de52e4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23215729/93e15e20-3b7c-4cae-8f8f-cc9c7fe63a2f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/23215729/5848437d-77f7-44bc-b8d9-c9d747c9690e

 
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/23215729/61b126ca-5e65-4e8e-8007-36e20f1ce20b

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23215729/cc197bb8-1418-4904-be5d-94aef04a8176

</details>

### CODE REVIEW

File: src/components/MentionSuggestions.tsx [hunk 1]
```diff
@@ -98,39 +89,37 @@ function MentionSuggestions(props) {
...
-                                </Text>
-                            ),
-                    )}
+                    {styledHandle &&
+                        styledHandle.map(
+                            ({text, isColored}) =>
+                                text !== '' && (
+                                    <Text
+                                        key={`${text}${isColored}`}
+                                        style={[StyleUtils.getColoredBackgroundStyle(isColored), styles.mentionSuggestionsHandle, {...(isColored && {color: themeColors.text})}]}
```
Comment (fabioh8010) [inline src/components/MentionSuggestions.tsx:98]: Why did you change `styles.text` to `themeColors.text`?

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:98]: we don't have such property anymore, changed to the property with same value

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: removed, since it never exists

File: src/components/MentionSuggestions.tsx [hunk 2]
```diff
@@ -98,39 +89,37 @@ function MentionSuggestions(props) {
...
-                                    style={[StyleUtils.getColoredBackgroundStyle(isColored), styles.mentionSuggestionsHandle, {...(isColored && {color: styles.text})}]}
-                                >
-                                    {text}
-                                </Text>
-                            ),
-                    )}
+                    {styledHandle &&
+                        styledHandle.map(
+                            ({text, isColored}) =>
+                                text !== '' && (
```
Comment (barttom) [inline src/components/MentionSuggestions.tsx:95]: or maybe `!!text`

File: src/components/AvatarTypes.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import CONST from '@src/CONST';
+
+type AvatarFunctionType = () => void;
+
+type AvatarType = {
```
Comment (barttom) [inline src/components/AvatarTypes.ts:5]: I'm not sure about this name. Looks like You use it in an icons attribute, so maybe AvatarIcon is more accurate? Also, the file name and variable are not consistent 'AvatarTypes' vs 'AvatarType'

File: src/styles/styles.ts [hunk 4]
```diff
@@ -438,10 +421,6 @@ const styles = (theme: ThemeDefault) =>
             color: theme.link,
         },
 
-        textUppercase: {
-            textTransform: 'uppercase',
-        },
-
```
Comment (barttom) [inline src/styles/styles.ts:444]: why did You remove this?

File: src/components/MentionSuggestions.tsx [hunk 5]
```diff
@@ -85,9 +76,9 @@ function MentionSuggestions(props) {
                     style={[styles.mentionSuggestionsText, styles.flexShrink1]}
                     numberOfLines={1}
                 >
-                    {_.map(styledDisplayName, ({text, isColored}, i) => (
+                    {styledDisplayName.map(({text, isColored}) => (
                         <Text
-                            key={`${text}${i}`}
+                            key={`${text}${isColored}`}
```
Comment (barttom) [inline src/components/MentionSuggestions.tsx:81]: Are you sure this is a proper key? Looks like adding index avoids duplicates

File: src/components/AvatarTypes.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import CONST from '@src/CONST';
+
+type AvatarFunctionType = () => void;
```
Comment (kubabutkiewicz) [inline src/components/AvatarTypes.ts:3]: From our guideline we should use suffix `Type` for types

File: src/components/AvatarTypes.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+type AvatarType = {
+    source: string | AvatarFunctionType;
+    type: typeof CONST.ICON_TYPE_AVATAR | typeof CONST.ICON_TYPE_WORKSPACE;
+    name: string;
+    id: number | string;
+    fallbackIcon: string | AvatarFunctionType;
```
Comment (kubabutkiewicz) [inline src/components/AvatarTypes.ts:10]: I think fallbackIcon should be optional

File: src/components/MentionSuggestions.tsx (hunk 2 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: Changed to Boolean

File: src/components/MentionSuggestions.tsx (hunk 5 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: reverted

File: src/components/AvatarTypes.ts (hunk 3 above)
Comment (pasyukevich) [reply src/components/AvatarTypes.ts]: renamed

File: src/components/MentionSuggestions.tsx [hunk 8]
```diff
@@ -1,73 +1,64 @@
...
+    measureParentContainer: () => void;
 };
 
 /**
  * Create unique keys for each mention item
- * @param {Object} item
- * @param {Number} index
- * @returns {String}
+ * @param item
+ * @param index
```
Comment (blazejkustra) [inline src/components/MentionSuggestions.tsx:49]: You can remove it

File: src/components/AvatarIconType.ts
Comment (blazejkustra) [inline src/components/AvatarIconType.ts:1]: For the future, don't name files with Type suffix 😄

File: src/components/AvatarIconType.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import CONST from '@src/CONST';
+
+type AvatarFunction = () => void;
+
+type AvatarIcon = {
```
Comment (blazejkustra) [inline src/components/AvatarIconType.ts:5]: Define these types directly in `MentionSuggestion` file, we can move it to a separate file as a follow-up.

File: src/components/MentionSuggestions.tsx [hunk 10]
```diff
@@ -1,73 +1,64 @@
...
-const keyExtractor = (item) => item.alternateText;
+const keyExtractor = (item: Mention) => item.alternateText;
 
-function MentionSuggestions(props) {
+function MentionSuggestions({prefix, mentions, highlightedMentionIndex = 0, onSelect, isMentionPickerLarge, measureParentContainer = () => {}}: MentionSuggestionsProps) {
     /**
      * Render a suggestion menu item component.
-     * @param {Object} item
-     * @returns {JSX.Element}
+     * @param item
```
Comment (blazejkustra) [inline src/components/MentionSuggestions.tsx:56]: Same here

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Comment by blazejkustra** [comment]
I think you can remove [NO QA] from the title and try add some QA Steps, thanks! 😄

File: src/styles/StyleUtils.ts [hunk 11]
```diff
@@ -1012,7 +1012,7 @@ function getAutoCompleteSuggestionContainerStyle(itemsHeight: number): ViewStyle
 /**
  * Select the correct color for text.
  */
-function getColoredBackgroundStyle(isColored: boolean): ViewStyle {
+function getColoredBackgroundStyle(isColored: boolean): StyleProp<TextStyle> | undefined {
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/styles/StyleUtils.ts]: You are right, looks like I had problems without the undefined because of the local cache

[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/styles/StyleUtils.ts]: Updated

File: src/components/MentionSuggestions.tsx [hunk 12]
```diff
@@ -98,39 +90,38 @@ function MentionSuggestions(props) {
...
-                                <Text
-                                    key={`${text}${i}`}
-                                    style={[StyleUtils.getColoredBackgroundStyle(isColored), styles.mentionSuggestionsHandle, {...(isColored && {color: styles.text})}]}
-                                >
-                                    {text}
-                                </Text>
-                            ),
-                    )}
+                    {styledHandle &&
+                        styledHandle.map(
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: changed

File: src/components/MentionSuggestions.tsx (hunk 8 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: done

File: src/components/AvatarIconType.ts
Comment (pasyukevich) [reply src/components/AvatarIconType.ts:1]: Totally agree with you, in this case, I didn't find any other way to create a type file in the component folder with the identification that it is a type, not a component

File: src/components/AvatarIconType.ts (hunk 9 above)
Comment (pasyukevich) [reply src/components/AvatarIconType.ts]: moved

File: src/components/MentionSuggestions.tsx (hunk 10 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: removed

**Comment by pasyukevich** [comment]
@blazejkustra PR updated

**Review by blazejkustra — APPROVED** [review]
Looks good, before making it ready for review someone from CK should review this Pr as well.

**Comment by blazejkustra** [comment]
> I think you can remove [NO QA] from the title and try add some QA Steps, thanks! 😄

Ah also, don't forget to add the steps

File: src/components/MentionSuggestions.tsx [hunk 13]
```diff
@@ -1,73 +1,71 @@
...
 import CONST from '@src/CONST';
 import AutoCompleteSuggestions from './AutoCompleteSuggestions';
 import Avatar from './Avatar';
-import avatarPropTypes from './avatarPropTypes';
 import Text from './Text';
 
-const propTypes = {
-    /** The index of the highlighted mention */
-    highlightedMentionIndex: PropTypes.number,
+// TODO: remove when avatarPropTypes will be migrated to TS
```
Comment (kubabutkiewicz) [inline src/components/MentionSuggestions.tsx:12]: should we keep this comment?

File: src/components/MentionSuggestions.tsx [hunk 14]
```diff
@@ -1,73 +1,71 @@
...
+type AvatarFunction = () => void;
 
-    /** Array of suggested mentions */
-    mentions: PropTypes.arrayOf(
-        PropTypes.shape({
-            /** Display name of the user */
-            text: PropTypes.string,
+type AvatarIcon = {
```
Comment (kubabutkiewicz) [inline src/components/MentionSuggestions.tsx:15]: in OnyxCommon module we have already Icon component which have same props , so you can use it instead of creating this type

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:15]: Not sure that I can use it 

These types are different in fields and types

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:15]: I've defined the type with the same signature that was used before (avatarPropTypes.js)

Comment (kubabutkiewicz) [reply src/components/MentionSuggestions.tsx:15]: I am almost certain that every Icon which we use in the app should be of type which you created, so I would adjust type of the Icon in OnyxCommon

File: src/components/MentionSuggestions.tsx (hunk 13 above)
Comment (kubabutkiewicz) [reply src/components/MentionSuggestions.tsx:12]: also I belive that its not the correct type.
the `source` prop should be a type of `string | React.FC<SvgProps>`

File: src/components/MentionSuggestions.tsx (hunk 14 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:15]: As I see we do not use this type from the OnyxCommon anywhere and it looks like we will have this type under types/onyx/avatar or smth like that

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:15]: @blazejkustra what do you think? 

I see that you authored this commit with Icon type creation in the OnyxCommon

File: src/components/MentionSuggestions.tsx (hunk 13 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:12]: It was added according to the propTypes file 

image

Comment (blazejkustra) [reply src/components/MentionSuggestions.tsx:12]: PropTypes aren't always correct, you have to check the usage

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:12]: I believe that further extensions should be part of Avatar component migration / propTypess migration

File: src/components/MentionSuggestions.tsx (hunk 14 above)
Comment (blazejkustra) [reply src/components/MentionSuggestions.tsx:15]: > As I see we do not use this type from the OnyxCommon anywhere and it looks like we will have this type under types/onyx/avatar or smth like that

It's used in Report.ts

After digging in the code I found where suggestions are created - [here](https://github.com/pasyukevich/expensify-app/blob/0138d943fc97c4576e41cb0fd937bb8b7d7bf2d0/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L141) and [here](https://github.com/pasyukevich/expensify-app/blob/0138d943fc97c4576e41cb0fd937bb8b7d7bf2d0/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L166).

I think you should be able to adjust the type in OnyxCommon like Kuba suggested

File: src/components/MentionSuggestions.tsx [hunk 15]
```diff
@@ -1,73 +1,71 @@
...
+    source: string | AvatarFunction;
+    type: typeof CONST.ICON_TYPE_AVATAR | typeof CONST.ICON_TYPE_WORKSPACE;
+    name: string;
+    id: number | string;
+    fallbackIcon?: string | AvatarFunction;
```
Comment (blazejkustra) [inline src/components/MentionSuggestions.tsx:20]: I think both source and fallbackIcon should be of type `string | React.FC<SvgProps>`

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/MentionSuggestions.tsx (hunk 13 above)
Comment (blazejkustra) [reply src/components/MentionSuggestions.tsx:12]: You can also debug it a little, console.log what is the type actually. Have you tried it?

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:12]: I didn't see messages in this thread except the first two, but now it looks like dialogue :joy:

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:12]: I believed before that a deep investigation for types that are not related directly should be part of the directly related component migration (Avatar migration in this case)

File: src/components/MentionSuggestions.tsx (hunk 15 above)
Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx]: changed

**Comment by pasyukevich** [comment]
@kubabutkiewicz @blazejkustra PR updated

**Review by blazejkustra — APPROVED** [review]

**Review by kubabutkiewicz — APPROVED** [review]
lgtm 😄

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by mollfpr — APPROVED** [review]
LGTM 👍

File: src/components/MentionSuggestions.tsx [hunk 16]
```diff
@@ -85,8 +73,9 @@ function MentionSuggestions(props) {
                     style={[styles.mentionSuggestionsText, styles.flexShrink1]}
                     numberOfLines={1}
                 >
-                    {_.map(styledDisplayName, ({text, isColored}, i) => (
+                    {styledDisplayName.map(({text, isColored}, i) => (
```
Comment (MonilBhavsar) [inline src/components/MentionSuggestions.tsx:76]: If it is undefined
```suggestion
                    {styledDisplayName?.map(({text, isColored}, i) => (
```

File: src/components/MentionSuggestions.tsx [hunk 17]
```diff
@@ -98,13 +87,13 @@ function MentionSuggestions(props) {
...
-                    {_.map(
-                        styledHandle,
+                    {styledHandle?.map(
                         ({text, isColored}, i) =>
-                            text !== '' && (
+                            Boolean(text) && (
                                 <Text
+                                    // eslint-disable-next-line react/no-array-index-key
                                     key={`${text}${i}`}
-                                    style={[StyleUtils.getColoredBackgroundStyle(isColored), styles.mentionSuggestionsHandle, {...(isColored && {color: styles.text})}]}
```
Comment (MonilBhavsar) [inline src/components/MentionSuggestions.tsx:107]: Why we're removing this style?

Comment (pasyukevich) [reply src/components/MentionSuggestions.tsx:107]: We never had a definition of this style

**Review by MonilBhavsar — COMMENTED** [review]
@pasyukevich could you please update PR author checklist. Looks like it is updated and hence the check is failing

**Comment by pasyukevich** [comment]
updated with the latest main and solved conflicts

**Review by MonilBhavsar — APPROVED** [review]
Looks good, thanks!
